### PR TITLE
feat(catalog-backend-module-azure): add Azure DevOps SCM event translation layer

### PR DESCRIPTION
**Pull request by lokeshkaki**
## Hey, I just made a Pull Request!

This PR adds Azure DevOps SCM event support to the Backstage catalog, following the request in Issue #32833.

It introduces a webhook analyzer that turns Azure DevOps `git.push` and `git.repo.*` payloads into generic catalog SCM events, so entities are refreshed as soon as files change, automatically unregistered when catalog files are removed, and seamlessly re-registered when files or repositories move. The new `AzureDevOpsScmEventsBridge` hooks this analyzer into the catalog module’s lifecycle, and the analyzer is exposed through the module’s public API.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
  - Added `.changeset/thin-lies-deliver.md` with a patch bump for `@backstage/plugin-catalog-backend-module-azure`
- [x] Added or updated documentation
  - Inline code comments document payload field mappings and edge cases
- [x] Tests for new functionality and regression tests for bug fixes
  -  Unit tests cover `git.push` file add, edit, delete, and rename scenarios; `git.repo.*` lifecycle events; unsupported event types; and malformed payload validation. Module wiring is covered in the existing module test.
- [ ] Screenshots attached (for UI changes)
  - backend-only change
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend-module-azure/src/module/catalogModuleAzureDevOpsEntityProvider.ts [hunk 1]
```diff
@@ -19,10 +19,13 @@ import {
   createBackendModule,
 } from '@backstage/backend-plugin-api';
 import { catalogProcessingExtensionPoint } from '@backstage/plugin-catalog-node';
+import { catalogScmEventsServiceRef } from '@backstage/plugin-catalog-node/alpha';
+import { eventsServiceRef } from '@backstage/plugin-events-node';
 import {
```
[the remark this answers is not in the document]
Comment (lokeshkaki) [reply plugins/catalog-backend-module-azure/src/module/catalogModuleAzureDevOpsEntityProvider.ts:24]: fixed
